Logging/LogHandler/FileLogHandler.cs

```diff
@@ -97,6 +158,56 @@ protected override void DisposeSelf()
                 );
 
             m_Writer.WriteLine($"{timestamp}{logLevel}{context}{message}");
+
+            if (m_Rotate && m_Writer.BaseStream.Length > RotateFileSizeLimit)
+            {
+                this.RotateFiles();
+            }
```

Comment:
Do we need to worry about validating values like `RotateFileSizeLimit`, to ensure it's positive or even a reasonable minimum value? Or what you set is what you get?

Reply:
I think we could `Debug.Assert` that it's at least greater than 0 if changing the type to `uint` is weird because of the API etc. 

If you set it to `int.MaxValue` though, that's on you. I don't think we should stop anyone because that probably means they don't care about diskspace and just wanted the "infinite" default which we don't have. It's doubtful they actually meant to rotate above `int.MaxValue`

Reply:
`int.MaxValue` is fine, but I meant if they set the file size limit to like... 5 (bytes) they're going to get a lot of log files. But maybe that's fine.

I'll just add an assert for `> 0`

Reply:
Yeah I think it's on them. The rotate limit will catch most of the damage from exploding amount of files and they'll just thrash the IO. Which hopefully they notice via profiling. I think it would be too difficult to assume a minimum otherwise.